BEST_PRACTICES.md

```diff
@@ -0,0 +1,146 @@
+
+#### Core
+- Core fixtures are framework-level fixtures that serve as setup and teardown for component tests.
+- Several special fixture modules deserve mention here:
+    - `broker.py`: Contains all the `target_sat` fixtures, scoped by usage, which should be used in every functional test in `robottelo`. These fixtures return the same satellite instance for all scoped fixtures, provided that the calling test does not have a destructive marker. For tests with destructive markers, they return a new satellite instance from an image.
+    - `contenthosts.py`: Houses all content host creation fixtures. All fixture markers for content hosts should be added here since the `fixture_markers` pytest plugin operates on these fixtures, adding markers and user properties to tests that use them.
+    - `sat_cap_factory.py`: Contains fixtures that create satellites/capsules from scratch without using the xDist satellite. These fixtures should be placed here, as the `factory_collection` pytest plugin operates on them, adding markers to tests that use these fixtures.
+    - `xdist.py`: Includes the xDist distribution session-scoped fixture, which distributes and generates satellites for running Robottelo tests. The behavior of xDist is based on the `xdist_behavior` set in the `server.yaml` configuration file.
+
+### Robottelo
+
+#### host_helpers
+- `api_factory.py`
+  - This module contains the `APIFactory` class. Helpers within this class facilitate integrated operations between components of the satellite object using satellite API calls.
+  - Since we favor API test setups, this class is the preferred location for adding generic helpers that can be used across UI, CLI, and API tests.
+  - Helpers added to this class or inherited by this class can be accessed as `sat_obj.api_factory.helper_func_name()`.
+
+- `cli_factory.py`
+  - This module contains the `CLIFactory` class. Helpers within this class enable integrated operations between components of the satellite object using satellite CLI hammer commands.
+  - While it is not our primary choice to add more CLI-based helpers, this class is the appropriate place to add them if the test demands it.
+  - Helpers added to this class or inherited by this class can be accessed as `sat_obj.cli_factory.helper_func_name()`.
+
+- `ui_factory.py`
+  - This module contains a `UIFactory` class that accepts a session object as a parameter. Helpers within this class enable integrated operations between components of the satellite object using the satellite UI.
+  - Although adding more UI-based helpers is not our preference, this class is suitable for adding them if the test demands it.
+  - Helpers added to this class or inherited by this class can be accessed as `sat_obj.ui_factory(session).helper_func_name()`.
+  - The session object provided here is the existing session from the test case where the helper is being accessed.
```

Comment:
based on our current conversations, it would likely be good to add a line about passing a different session object into the helper when needed.